### PR TITLE
Allow template-haskell-2.22; extend CI to GHC 9.10

### PR DESCRIPTION
**Pull request by andreasabel**
CI passes: https://github.com/andreasabel/tagged/actions/runs/8308675525/job/22739167225?pr=1

Published as revision:

2024-03-16T15:19:10Z  	AndreasAbel	  [tagged-0.8.8-r1](https://hackage.haskell.org/package/tagged-0.8.8/revisions)

### CODE REVIEW

File: tagged.cabal [hunk 1]
```diff
@@ -14,26 +14,29 @@ description:    Haskell 98 phantom types to avoid unsafely passing dummy argumen
...
-             , GHC == 9.0.2
-             , GHC == 9.2.8
-             , GHC == 9.4.5
-             , GHC == 9.6.2
+tested-with:
+  -- GHC == 7.0.4
+  -- GHC == 7.2.2
+  -- GHC == 7.4.2
+  -- GHC == 7.6.3
+  -- GHC == 7.8.4
```
Comment (RyanGlScott) [inline tagged.cabal:22]: Why does this change comment out these versions? (Perhaps there is a motivation here, but it's not explained in the commit message.)

Comment (andreasabel) [reply tagged.cabal:22]: My patched version of `haskell-ci` defaults to `distribution: focal` (20.04) and `ghcup` as installation method, thus, does not support these older ghc versions out of the box.

If you wish to keep them tested by CI, I can switch to `distribution: bionic` (18.04) and restore them.

Comment (andreasabel) [reply tagged.cabal]: I restored CI for GHC 7

Comment (RyanGlScott) [reply tagged.cabal]: Ah, I see. I'm content to leave it on Ubuntu 18.04 for now, but I suppose we will need to address this at some point or another. In any case, that's best left for a different PR.

**Review by RyanGlScott — APPROVED** [review]

**Pull request merged**